### PR TITLE
Include TRex Python API to the Fabric-TNA PTF test container image

### PR DESCRIPTION
**Pull request by dariusgrassi**
[SDFAB-232] Updating Dockerfile to install dependencies for TRex Python API to PTF testvectors container image

### CODE REVIEW

File: tools/ptf/Dockerfile [hunk 1]
```diff
@@ -95,9 +98,27 @@ ENV PIP_DEPS \
 RUN apt update && apt install -y $RUNTIME_DEPS
 RUN pip3 install --no-cache-dir --root /python_output $PIP_DEPS
 
+# Install TRex deps
+FROM alpine:3.12.1 as builder
```
Comment (Yi-Tseng) [inline tools/ptf/Dockerfile:102]: Suggest using `trex-builder`

File: tools/ptf/Dockerfile [hunk 2]
```diff
@@ -106,12 +127,20 @@ ENV RUNTIME_DEPS \
...
 
+ENV TREX_EXT_LIBS=${TREX_EXT_LIBS}
+ENV PYTHONPATH=/workspace/trex-scripts:${TREX_EXT_LIBS}:${TREX_LIBS}
+COPY --from=builder /output /
 COPY --from=proto-deps /output /usr/lib/python3.8/dist-packages
 COPY --from=ptf-deps /python_output /
 RUN ldconfig
 
-ENTRYPOINT []
+ENTRYPOINT []
```
Comment (Yi-Tseng) [inline tools/ptf/Dockerfile:146]: keep the new line in the end of the file

File: tools/ptf/Dockerfile (hunk 1 above)
Comment (dariusgrassi) [reply tools/ptf/Dockerfile]: Added

File: tools/ptf/Dockerfile (hunk 2 above)
Comment (dariusgrassi) [reply tools/ptf/Dockerfile]: Fixed!

File: tools/ptf/Dockerfile [hunk 3]
```diff
@@ -106,10 +127,18 @@ ENV RUNTIME_DEPS \
...
     netbase
 
 RUN apt update && apt install -y $RUNTIME_DEPS
+RUN pip3 install scipy==1.5.4 numpy==1.19.4 matplotlib==3.3.3
 
+ENV TREX_EXT_LIBS=${TREX_EXT_LIBS}
+ENV PYTHONPATH=/workspace/trex-scripts:${TREX_EXT_LIBS}:${TREX_LIBS}
+COPY --from=trex-builder /output /
 COPY --from=proto-deps /output /usr/lib/python3.8/dist-packages
 COPY --from=ptf-deps /python_output /
```
Comment (dariusgrassi) [inline tools/ptf/Dockerfile:143]: Will these COPY statements overwrite directories in the container?

**Review by Yi-Tseng — APPROVED** [review]
LGTM, I tested both fabric-tna test and current flirt test and they both works
feel free to merge it

**Pull request merged**